### ISSUE TITLE
"Compare database objects": missing target directory causes "Invalid Thread Access"

### ISSUE THREAD

**Issue by 2019-05-10** [dbeaver/dbeaver#12345]
#### System information: 
- Operating system (distribution) and version: Debian/SID
- DBeaver version: 21.0.4.202105021514
- Additional extensions

#### Connection specification:
- Database name and version
- Driver name
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 

#### Describe the problem you're observing:

Comparing two databases (Compare/Migrate -> Simple Structure Compare)
Output Type: Save to File
Directory: /foo/bar

If the directory does not exist, an "Invalid Thread Access" error is thrown.
![Screenshot_20210504_121440](https://user-images.githubusercontent.com/50481865/116989712-67d42c80-acd2-11eb-89d8-2cc2219b9002.png)
a) the error message is misleading and rather pointless -- from the description one can hardly understand what exactly goes wrong
b) directory's existence should be checked beforehand, and either be silently created if missing or a clear and helpful message should appear

Stack trace:

```
!ENTRY org.jkiss.dbeaver.model 4 0 2021-05-04 12:04:51.617
!MESSAGE Invalid thread access
!STACK 0
org.eclipse.swt.SWTException: Invalid thread access
at org.eclipse.swt.SWT.error(SWT.java:4889)
at org.eclipse.swt.SWT.error(SWT.java:4804)
at org.eclipse.swt.SWT.error(SWT.java:4775)
at org.eclipse.swt.widgets.Widget.error(Widget.java:555)
at org.eclipse.swt.widgets.Widget.checkWidget(Widget.java:471)
at org.eclipse.swt.widgets.Label.getText(Label.java:401)
at org.eclipse.jface.dialogs.TitleAreaDialog.updateMessage(TitleAreaDialog.java:614)
at org.eclipse.jface.dialogs.TitleAreaDialog.setErrorMessage(TitleAreaDialog.java:408)
at org.eclipse.jface.wizard.WizardDialog.updateMessage(WizardDialog.java:1378)
at org.eclipse.jface.wizard.WizardPage.setErrorMessage(WizardPage.java:254)
at org.jkiss.dbeaver.tools.compare.simple.ui.CompareObjectsWizard.showError(CompareObjectsWizard.java:94)
at org.jkiss.dbeaver.tools.compare.simple.ui.CompareObjectsWizard.renderReport(CompareObjectsWizard.java:176)
at org.jkiss.dbeaver.tools.compare.simple.ui.CompareObjectsWizard.access$1(CompareObjectsWizard.java:147)
at org.jkiss.dbeaver.tools.compare.simple.ui.CompareObjectsWizard$1.run(CompareObjectsWizard.java:113)
at org.jkiss.dbeaver.ui.UIUtils.lambda$11(UIUtils.java:1578)
at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:122)
```

**Comment by kseniiaguzeeva** [comment]
Thank you for the report.

**Comment by LonwoLonwo** [comment]
Fixed in DBeaver 21.1.5

**Comment by Matvey16** [comment]
Verified
